egs2/TEMPLATE/s2t1/s2t.sh

```diff
@@ -0,0 +1,1853 @@
+            "${data_feats}/${dset}"
+    done
+
+    if [ -n "${post_process_local_data_opts}" ]; then
+        # Do any additional local data post-processing here
+        local/data.sh ${post_process_local_data_opts} --s2t_data_dir "${data_feats}/${train_set}"
+    fi
+
+    # shellcheck disable=SC2002,SC2068,SC2005
+    for lm_txt in ${lm_train_text[@]}; do
+        suffix=$(echo "$(basename ${lm_txt})" | sed 's/text//')
+        <${lm_txt} awk -v suffix=${suffix} ' { if( NF != 1 ) {$1=$1 suffix; print $0; }} '
+    done > "${data_feats}/lm_train.txt"
+fi
+
+
+if [ ${stage} -le 5 ] && [ ${stop_stage} -ge 5 ] && ! [[ " ${skip_stages} " =~ [[:space:]]5[[:space:]] ]]; then
+    if [ "${token_type}" = bpe ]; then
+        log "Stage 5: Generate token_list from ${bpe_train_text} using BPE"
+
+        mkdir -p "${bpedir}"
+        # shellcheck disable=SC2002
+        cat ${bpe_train_text} | cut -f 2- -d" "  > "${bpedir}"/train.txt
+
+        if [ -n "${bpe_nlsyms}" ]; then
+            if test -f "${bpe_nlsyms}"; then
+                bpe_nlsyms_list=$(awk '{print $1}' ${bpe_nlsyms} | paste -s -d, -)
+                _opts_spm="--user_defined_symbols=${bpe_nlsyms_list}"
+            else
+                _opts_spm="--user_defined_symbols=${bpe_nlsyms}"
+            fi
+        else
+            _opts_spm=""
+        fi
+
+        spm_train \
+            --input="${bpedir}"/train.txt \
+            --vocab_size="${nbpe}" \
+            --model_type="${bpemode}" \
+            --model_prefix="${bpeprefix}" \
+            --character_coverage=${bpe_char_cover} \
+            --input_sentence_size="${bpe_input_sentence_size}" \
+            ${_opts_spm}
+
+        {
+            echo "${blank}"
+            echo "${oov}"
+            # Remove <unk>, <s>, </s> from the vocabulary
+            <"${bpeprefix}".vocab awk '{ if( NR != 1 && NR != 2 && NR != 3 ){ print $1; } }'
+            echo "${sos}"
+            echo "${eos}"
+            echo "${sop}"
+        } > "${token_list}"
+
+    elif [ "${token_type}" = char ] || [ "${token_type}" = word ]; then
+        log "Stage 5: Generate character level token_list from ${lm_train_text}"
+
+        _opts="--non_linguistic_symbols ${nlsyms_txt}"
+
+        # The first symbol in token_list must be "<blank>" and the last must be also sos/eos:
+        # 0 is reserved for CTC-blank for ASR and also used as ignore-index in the other task
+        ${python} -m espnet2.bin.tokenize_text  \
+            --token_type "${token_type}" \
+            --input "${data_feats}/lm_train.txt" --output "${token_list}" ${_opts} \
+            --field 2- \
+            --cleaner "${cleaner}" \
+            --g2p "${g2p}" \
+            --write_vocabulary true \
+            --add_symbol "${blank}:0" \
+            --add_symbol "${oov}:1" \
+            --add_symbol "${sop}:-1" \
+            --add_symbol "${eos}:-2" \
+            --add_symbol "${sos}:-3"
```

Comment:
Is there a specific reason that you use -1, -2, and -3. I remember the default for padding is set as -1, not sure that would be affected.

Reply:
I just want to make them at the end of the vocab. In the original ASR, we only use one symbol for SOS and EOS, which is inserted at the end (-1).

Actually -1 means the last element. The token int is always positive, so it should not affect padding. Please correct me.

Reply:
My apology, it was my mistake.